packages/web3-utils/src/socket_provider.ts

```diff
@@ -288,8 +289,10 @@ export abstract class SocketProvider<
 			this.isReconnecting = false;
 			this._clearQueues();
 			this._removeSocketListeners();
-			const errorMsg = `Max connection attempts exceeded (${this._reconnectOptions.maxAttempts})`;
-			this._eventEmitter.emit('error', errorMsg);
+			this._eventEmitter.emit(
+				'error',
```

Comment:
https://github.com/web3/web3.js/runs/11848321915 codecov seems to be failing, otherwise PR LGTM

Reply:
Thanks @luu-alex ,
Actually, this is tested twice inside web3-providers-ws and web3-providers-ipc. But CodeCov does not check across packages. So, I think it is safe to ignore it. Or what do you think?

Reply:
couldn't we have a dedicated test in `web3-utils` for that?

Reply:
Actually, we can. But it will require a complex scenario that is already tested twice with IpcProvider and WebSocketProvider.  So, writing the test will be just time-consuming, I think :blush: .